src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -1264,6 +1264,62 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    void TerminalPage::_HandleSaveSnippet(const IInspectable& /*sender*/,
+                                       const ActionEventArgs& args)
+    {
+        if constexpr (!Feature_SaveSnippet::IsEnabled())
+        {
+            return;
+        }
+
+        if (args)
+        {
+            if (const auto& realArgs = args.ActionArgs().try_as<SaveSnippetArgs>())
+            {
+                if (realArgs.Commandline().empty())
+                {
+                    if (const auto termControl{ _GetActiveControl() })
+                    {
+                        if (termControl.HasSelection())
+                        {
+                            const auto selections{ termControl.SelectedText(true) };
+                            const auto selection = std::accumulate(selections.begin(), selections.end(), std::wstring());
+                            realArgs.Commandline(selection);

```

Comment:
This shouldn't modify the `realArgs.Commandline()` but instead extract the hstring and then modify the local reference. Like this:

```cpp
auto commandline = realArgs.Commandline();
if (commandline.empty() {
    // ...
    commandline = selection;
}

if (commandline.empty()) {
    // ...
}
```

Reply:
updated